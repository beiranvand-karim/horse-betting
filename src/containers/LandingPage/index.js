import React, {Component} from "react"
import SearchGame from "../../components/SearchGame"
import config from "../../config"
import Error from "../../components/Error"
import Loading from "../../components/Loading"
import games from "../../backend/games"
import Game from "../../components/Game"

export class LandingPage extends Component {
   constructor(props) {
      super(props);
      this.mockBackend = this.mockBackend.bind(this);
      this.fetchData = this.fetchData.bind(this);
      this.searchGameFormSubmit = this.searchGameFormSubmit.bind(this);
      this.change = this.change.bind(this);
      window.fetch = this.mockBackend
   }

   state = {
      data: null,
      loading: false,
      error: null,
      gameId: ''
   };

   mockBackend() {
      return new Promise((resolve) => {
         setTimeout(() => resolve(games), 1000)
      })
   }

   searchGameFormSubmit(e) {
      e.preventDefault();
      this.fetchData('V4');
   }

   change(e) {
      this.setState({
         gameId: e.target.value
      })
   };

   fetchData(gameId) {
      this.setState({loading: true});
      fetch(`${config.server}/services/racinginfo/v1/api/products/${gameId}`, {
         method: "GET",
         headers: {
            "Content-Type": "application/json"
         }
      })
         .then(data => {
            if (!data.status) {
               this.setState({loading: false});
               this.setState({data})
            } else {
               this.setState({loading: false});
               this.setState({error: data.error})
            }
         })
         .catch(error => {
            this.setState({loading: false});
            this.setState({error})
         })
   }

   render() {
      const {gameId, error, loading, data} = this.state;
      if (error) {
         return <Error error={error}/>
      }

      if (loading) {
         return <Loading/>
      }

      return <div>
         <h1>landing page</h1>
         <SearchGame gameId={gameId} change={this.change} searchGameFormSubmit={this.searchGameFormSubmit}/>
         {
            (data) && (<Game data={data}/>)
         }
      </div>
   }
}

export default LandingPage