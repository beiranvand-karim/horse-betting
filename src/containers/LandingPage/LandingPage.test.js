/* global describe, it, expect*/
import {shallow} from "enzyme"
import React from "react"
import LandingPage from "./"
import renderer from "react-test-renderer"
import {MemoryRouter} from "react-router-dom"

describe('<LandingPage/>', () => {
   it('should render without crashing', () => {
      shallow(<LandingPage/>)
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><LandingPage/></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });
});