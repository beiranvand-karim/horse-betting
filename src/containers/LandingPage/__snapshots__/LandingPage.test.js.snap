// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage/> should matches the snapshot 1`] = `
<div>
  <h1>
    landing page
  </h1>
</div>
`;
